pkg/util/kubernetes/hostinfo/node_labels.go

```diff
@@ -46,7 +51,12 @@ func GetNodeClusterNameLabel(ctx context.Context) (string, error) {
 	}
 
 	clusterNameLabels := []string{
-		"alpha.eksctl.io/cluster-name", // EKS cluster-name label
+		eksClusterNameLabelKey,       // EKS cluster-name label
+		datadogADClusterNameLabelKey, // Generic Datadog AD cluster-name label
+	}
+	// check if a node label has been added on the config
+	for _, label := range config.Datadog.GetStringSlice("kubernetes_node_labels_as_cluster_name") {
```

Comment:
I think the custom `kubernetes_node_labels_as_cluster_name` should be tried first.
The order should probably be: custom > Generic DD > EKS.

Also you have the order issue is the list of providers in `pkg/util/kubernetes/clustername/clustername.go`.
Currently setting `kubernetes_node_labels_as_cluster_name` will have not override an auto-detected cluster name. Not sure if that would be expected or not, in any case it deserves some documentation.

Reply:
I catch up this issue when building some unit-tests